apps/microsoft-teams/frontend/src/components/config/NotificationsSection/NotificationsSection.tsx

```diff
@@ -62,14 +62,18 @@ const NotificationsSection = (props: Props) => {
     const updatedNotification = { ...existingNotification, ...editedNotification };
     notificationsPayload[index] = updatedNotification;
 
+    // Use a Set to keep track of unique keys
+    const uniqueKeys = new Set();
```

Comment:
trying a compound key approach to avoid the heavy initial logic. @lilbitner this seems to also work as intended, will test it out further 🙇